### ISSUE TITLE
 used centernet for training and found unusual detection results

### ISSUE THREAD

**Issue by Bin-ze** [hu64/hughesperreault.com#1]
Hello, sorry to disturb you
I have observed that your spotnet warehouse is no longer public and there is no way to contact you, so I am sorry to post questions under this warehouse. If I disturb you, I am very sorry。

I used centernet to train the UAVDT dataset, but found some unusual problems
Want to ask you for help.
I used centernet to train the UAV data set, but found that half of a picture could not detect the target, and the other half of the picture could be detected very well, even though there is almost no difference in the characteristics of the two targets.
I checked the code carefully, but I didn’t find the problem. How can I think about solving such a problem?
The picture of the test results is as follows, almost all test results have such problems：
Send the image directly to the detector, the results are as follows：
![image](https://user-images.githubusercontent.com/72087870/148221853-76a7d52a-1837-4f10-994e-70d43d471009.png)
When I flip the image and send it to the detector, the result is as follows：
![image](https://user-images.githubusercontent.com/72087870/148222359-e9e5212e-771a-477d-9923-ab07d07c5f3f.png)
horizontal flip
![image](https://user-images.githubusercontent.com/72087870/148221918-c18f05b7-2253-48a7-8953-643fadeb25d5.png)
Flip vertically
![image](https://user-images.githubusercontent.com/72087870/148221935-0ebf728c-c754-4800-8478-9bf2d425d127.png)

When sampling, convolution is performed, and the convolution strokes every position of the image. This problem should not occur in theory, unless the convolution kernel does not extract features from the right side, but the result of flipping the image will get better. Even the opposite result appears. Some images on the left cannot be detected. Does this mean overfitting? I feel very strange. Such a result is something I have never encountered before. I did not change any settings of the centernet, but used other data. I am sure that my operation is normal because I have done a lot of research on mmdetection before。

**Comment by hu64** [comment]
Hi, you can use [this](https://1drv.ms/u/s!Ahf1HD3oV-4xgtwWZzq5bV34NIlcMA?e=NKxPBW) link for SpotNet models. Has for your other problem, my first intuition is that there might not be annotations for these examples, I would check that if I were you (the ignore regions...). In that case, the model would be overfitting indeed.  Best of luck in your research!

**Comment by Bin-ze** [comment]
Do you have a single-category json file, I observed that in your research, all categories are grouped into one category, I think this will affect the final result to some extent. I would appreciate it if a single category of json could be provided to me.
Would it make sense for me to process the json file you provided as a single category if you are not using such processing?

**Comment by Bin-ze** [comment]
I used the exact same json file as you, and the learning rate to train the UAV dataset on centernet, but the result is very different from your paper, under the validation of matlab tool, my best model is about the same as your result The difference is 17AP, I don't know why, very much looking forward to your reply

**Comment by hu64** [comment]
Hi, it is true that I grouped the categories for training the detectors, yes. In my opinion, this should do a big difference. That might be it? Otherwise I am not sure…

**Comment by Bin-ze** [comment]
I don't think this will affect performance, because you categorize all categories into one category, but my problem is that no matter how I do it, I can't achieve the performance in your thesis, which makes it impossible for me to make strong improvements. I don't know how to do it, and I find that using more than 2,000 photos of your training, network No matter how big it is, it may be difficult to achieve better results, because lightweight networks are enough to fit such small data. I don't know what to do.

Send your own iPhone

**Comment by hu64** [comment]
I no longer have access to my dev. computer since I graduated so I can’t really help you debug this… I few pointers: make sure to start training with pretrained weights, monitor the validation loss during training and use to best model, and finally make sure to test using full image resolutions. Beyond that you can make some tests using other datasets… best of luck!

**Comment by Bin-ze** [comment]
I did everything you said, but I couldn't run 51% of the accuracy with the centernet+hg backbone network, and the result I ran was only about 40%. I couldn't achieve that result anyway, so the improvement I made could not exceed the accuracy of your baseline, no Convincing, I'm very troubled by this problem. My improvement has been improved on the basis of my network, but the result of the improvement is not as good as that of your baseline, which makes me very frustrated.

Send your own iPhone

**Comment by hu64** [comment]
I am sorry I can’t help my you more than that. I remember seeing another paper with 50+ ap on uavdt, checkout https://arxiv.org/pdf/2101.12694.pdf. You might find some help in there? I also remember seeing improved centernet implementations on GitHub which I never tried, maybe have a go at them?

**Comment by Bin-ze** [comment]
Did you try TTA, or flip test when inferring the results?
I found that I used centernet for UAV dataset training, and the result was 40.38%, while your result shows 51.2%, a difference of 10 points. If TTA is not used, the huge gap seems to be completely inexplicable. I tried heavier nets and found that on this dataset it doesn't improve as you use heavier nets. I don't know how to do it, my improvement is getting a boost from my baseline, but my baseline result seems to be too low to be competitive with yours, even though I use the same network as yours, This makes my thesis unable to describe such a result, what should I do

**Comment by hu64** [comment]
No tta or flip no. An idea I just had is that you should and sure your scripts for converting formats back and forth is correct. You can make some basic tests by converting GT and then back, if you know what I mean. A small error of one pixel in the width could make a big difference. Visually do your results look good? And what about other datasets? Can you reproduce CenterNet results on other datasets?
